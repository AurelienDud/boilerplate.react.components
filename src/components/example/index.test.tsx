import React from 'react'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import Example from './index'

const defaultProps = {
  label: "hello"
}

/**
 * Enzyme rendering
 */
test('Test using Enzyme', () => {
  const component = shallow(
    <Example {...defaultProps} />
  )

  expect(component.text()).toEqual(defaultProps.label)
})

/**
 * Snapshot
 */
test('Test using snapshot', () => {
  const component = renderer.create(
    <Example {...defaultProps} />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})