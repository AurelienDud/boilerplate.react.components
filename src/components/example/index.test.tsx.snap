// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test using snapshot 1`] = `
<button
  className="go1776512497 go2993169732 go3862641954"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
  type="button"
>
  hello
</button>
`;
